javascript-linting-rules.md

```diff
@@ -0,0 +1,54 @@
+# Javascript Linting Rules
+
+The following rules should be applied to all of our javascript projects.
+
+_A project may use additional rules to ones listed here, but these would not be part of the current standard and such rules are subject to change in the future._
+
+## Current Rules
+
+These are stored in the [.eslintrc](https://raw.githubusercontent.com/holidayextras/culture/.eslintrc) in this repository.
+
+For an explanation of these rules please refer to the [ESLint documentation](http://eslint.org/docs/rules/).
+
+### TLDR
+
+For the lazy here is a brief description of the rules specified:
+
+1. `"quotes": 0`
+1. `"no-underscore-dangle": 0`  Allow variables beginning with `_`.
+1. `"curly": [2, "multi-line"]` Force curly braces except on single line statements.
+1. `"camelcase": [2, {"properties": "never"}]` Force camel case variables except on object properties.
+1. `"space-in-parens": [2, "never"]` Don't allow spaces inside round brackets and content.
```

Comment:
This is a subjective rule, but my reasoning for choosing it was that the majority of popular o/s javascript code uses the version without spaces, for example:

https://github.com/jashkenas/backbone/blob/master/backbone.js
https://github.com/jashkenas/underscore/blob/master/underscore.js
https://github.com/moment/moment/blob/develop/moment.js
https://github.com/substack/node-browserify/blob/master/index.js
https://github.com/facebook/react/blob/master/main.js

And other known styles:

https://github.com/airbnb/javascript
http://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml
https://github.com/feross/standard